pkg/controller/dnsendpoint/dnsendpoint_controller.go

```diff
@@ -145,10 +179,19 @@ func (r *ReconcileDNSEndpoint) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, nil
 	}
 
-	domain := instance.Spec.Zone
-	dnsLog = dnsLog.WithField("domain", domain)
+	fullDomain := instance.Spec.Zone
+	splitDomain := strings.Split(fullDomain, ".")
+	rootDomain := strings.Join(splitDomain[1:], ".")
```

Comment:
This is not a sufficient method of determining the root domain. You could have a full domain of `abc.xyz.example.com` linked to the root domain of `example.com`.

I suggest that, instead of storing the scrapers and queries in maps, we store them in a slice. To find the right one to use, the controller will need to iterate over all of the scrapers until it finds one that works for the domain.